runtime/compiler/control/JitDump.cpp

```diff
@@ -482,9 +482,96 @@ runJitdump(char *label, J9RASdumpContext *context, J9RASdumpAgent *agent)
 
    try
       {
-      // if our compinfo is null, we are an application thread
+      // Process user defined options first. The user can specify any method signature to be recompiled at an arbitrary
+      // dump event. This is typically used for debugging purposes so we process this option first if it exists.
+      if (agent->dumpOptions != NULL)
+         {
+         // TODO: This is a total hack. Surely there must be an API out there where we can supply a full method
+         // signature and get back a J9Method* if it exists? Right? I've searched but I cannot find this API. I
+         // will tag this in a review and hopefully someone knows the right answer here.
+         J9Method* userMethod = NULL;
+
+            {
+            TR::VMAccessCriticalSection findUserMethod(TR_J9VMBase::get(jitConfig, crashedThread));
+
+            char* indexOfDot = strchr(agent->dumpOptions, '.');
+            if (indexOfDot != NULL)
+               {
+               J9Class* userClass = crashedThread->javaVM->internalVMFunctions->internalFindClassUTF8(crashedThread,
+                  reinterpret_cast<U_8*>(agent->dumpOptions), indexOfDot - agent->dumpOptions, crashedThread->javaVM->systemClassLoader, J9_FINDCLASS_FLAG_EXISTING_ONLY);
+
+               if (userClass != NULL)
+                  {
+                  J9JNINameAndSignature nameAndSig;
+                  nameAndSig.name = indexOfDot + 1;
+
+                  char* indexOfParen = strchr(agent->dumpOptions, '(');
+                  if (indexOfParen != NULL)
+                     {
+                     nameAndSig.nameLength = indexOfParen - nameAndSig.name;
+                     nameAndSig.signature = indexOfParen;
+                     nameAndSig.signatureLength = strlen(nameAndSig.signature);
+
+                     userMethod = reinterpret_cast<J9Method*>(crashedThread->javaVM->internalVMFunctions->javaLookupMethod(crashedThread,
+                        userClass, reinterpret_cast<J9ROMNameAndSignature*>(&nameAndSig), NULL, J9_LOOK_JNI | J9_LOOK_NO_JAVA));
+                     }
+                  }
+               }
+            }
```

Comment:
Not sure who to tag here really, so I'll just make this self review and put the question in Slack. Hopefully someone will be able to help because I don't want us merging this hacky code (even though it does work).

Reply:
There's a fundamental problem with what you're trying to do. You can have several different J9Methods with the same signature but loaded by different class loaders. The only methods you are guaranteed to uniquely find with just a signature are methods of system classes (see `TR_J9VM::getMethodFromName`).

Reply:
Given that this is a debugging option used by developers/service, how often would you think we would encounter such a case? Would you say more often than not (95/5 split?) a method signature to class loader is unique?

Reply:
> Given that this is a debugging option used by developers/service, how often would you think we would encounter such a case? Would you say more often than not (95/5 split?) a method signature to class loader is unique?
>
> I'm willing to bet in a large majority of uses cases the signature to J9Method* will be unique.

Yeah I would imagine so; but I have no experience to say for sure. That said I do agree that having the wrong method traced doesn't leave the dev in a worse position.

Reply:
This should be fixed now in 1e3a722.